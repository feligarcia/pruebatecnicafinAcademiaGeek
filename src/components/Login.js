import React from "react";
import { useFormik } from "formik";
import { useDispatch } from "react-redux";
import "../styles/GoogleBtn.css";
import {
  loginGoogle,
  loginEmailPassword,
  loginFacebook,
} from "../redux/actions/actionLogin";

import { useNavigate } from "react-router-dom";
import { Button } from "react-bootstrap";

const Login = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const formik = useFormik({
    initialValues: {
      email: "",
      password: "",
    },
    onSubmit: (data) => {
      dispatch(loginEmailPassword(data));
      // navigate("/");
    },
  });

  const handleGoogle = () => {
    dispatch(loginGoogle());
    // navigate("/");
  };
  const handleFacebook = () => {
    dispatch(loginFacebook());
    // navigate("/");
  };

  return (
    <div className="divlog">
      <form className="form-group" onSubmit={formik.handleSubmit}>
        <label>Correo electrónico</label>

        <input
          id="inputEmailSignIn"
          type="email"
          className="form-control mt-2"
          name="email"
          autoComplete="off"
          placeholder="Tu correo"
          required
          onChange={formik.handleChange}
        />
        <label>Contraseña</label>

        <input
          id="inputpasswordSignIn"
          type="password"
          className="form-control mt-2"
          name="password"
          autoComplete="off"
          placeholder="Ingresa tu contraseña"
          required
          onChange={formik.handleChange}
        />

        <br></br>
        <div className="d-grid gap-2 mx-auto mt-2">
          <Button value="Save" type="submit" variant="outline-primary">
            Ingresar
          </Button>
        </div>
        <p>Puedes probar con prueba@prueba.com | 123456</p>
        <br></br>

        <div className="google-btn" onClick={handleGoogle}>
          <div className="google-icon-wrapper">
            <img
              className="google-icon"
              src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
              alt=""
            />
          </div>
          <p className="btn-text">
            <b>Entrar con google</b>
          </p>
        </div>
        <br></br>

        <img
          onClick={handleFacebook}
          className="face-icon"
          src="https://scontent.feoh4-3.fna.fbcdn.net/v/t39.2365-6/17639236_1785253958471956_282550797298827264_n.png?_nc_cat=105&ccb=1-5&_nc_sid=ad8a9d&_nc_eui2=AeGuNfqzKUEqyoe9TXtNuhyTDpRKUGl8lBMOlEpQaXyUE0eONNlAaXo9dvUfEaw_MjOGEu0OH1fUkcPG06d-uBV4&_nc_ohc=7OIt3P9C1a4AX8w0tH8&_nc_ht=scontent.feoh4-3.fna&oh=00_AT80QfKiSQ3t8hZWNvUeCROBcCLzim25co5kFo0nb4Xi_g&oe=62216A16"
          alt=""
        />
        <br></br>
        <br></br>
        <h6 onClick={() => navigate("/registro")} className="linklogintitle">
          {" "}
          ¿Deseas crear una cuenta? Haz click aqui
        </h6>
      </form>
    </div>
  );
};

export default Login;
