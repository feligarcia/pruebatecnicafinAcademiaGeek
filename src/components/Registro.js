import React, { useState } from "react";
import { useFormik } from "formik";
import { useDispatch } from "react-redux";
import { createUserActionAsincrono } from "../redux/actions/actionLogin";
import { imgUpload } from "../functions/imgUpload";
import { useNavigate } from "react-router-dom";
import { Button } from "react-bootstrap";

const Registro = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const [imageURL, setimageURL] = useState("");
  const formik = useFormik({
    initialValues: {
      name: "",
      email: "",
      password: "",
      image: "",
    },
    onSubmit: (data) => {
      data.image = imageURL;
      dispatch(createUserActionAsincrono(data));
    },
  });

  const handleFileChanged = (e) => {
    e.preventDefault();
    const file = e.target.files[0];
    imgUpload(file)
      .then((response) => {
        setimageURL(response);
      })
      .catch((error) => console.log(error));
  };

  return (
    <div className="divlog">
      <form className="form-group" onSubmit={formik.handleSubmit}>
        <label>Nombre</label>
        <input
          id="inputName"
          type="text"
          className="form-control mt-2"
          name="name"
          autoComplete="off"
          placeholder="Ingresa tu nombre"
          required
          onChange={formik.handleChange}
        />
        <label>Correo electrónico</label>
        <input
          id="inputEmail"
          type="email"
          className="form-control mt-2"
          name="email"
          autoComplete="off"
          placeholder="Tu correo"
          required
          onChange={formik.handleChange}
        />
        <label>Contraseña</label>
        <input
          id="inputpassword"
          type="password"
          className="form-control mt-2"
          name="password"
          autoComplete="off"
          placeholder="Configura una contraseña"
          required
          onChange={formik.handleChange}
        />
        <label>Imagen para tu perfil</label>
        <input
          id="inputImage"
          type="file"
          className="form-control "
          placeholder="Adjuntar una imagen"
          name="image"
          required
          onChange={handleFileChanged}
        />
        <br></br>
        <div className="d-grid gap-2 mx-auto mt-2">
          <Button value="Save" type="submit" variant="outline-primary">
            Registrar
          </Button>
        </div>
        <br></br>
        <h6 onClick={() => navigate("/login")} className="linklogintitle">
          ¿Ya tienes una cuenta? Haz click aqui
        </h6>
      </form>
    </div>
  );
};

export default Registro;
