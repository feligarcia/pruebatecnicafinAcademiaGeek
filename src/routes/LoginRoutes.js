import { Navigate } from "react-router-dom"

export const LoginRoutes=({isAuthenticated, children}) => {
    return !isAuthenticated
    ?children
    : <Navigate to="/"/>
}