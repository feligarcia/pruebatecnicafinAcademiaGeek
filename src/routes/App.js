import { BrowserRouter, HashRouter, Route, Routes } from "react-router-dom";
import "../styles/App.css";
import ListCard from "../containers/ListCard";
import Navbarapp from "../components/Navbarapp";
import CardDetail from "../components/CardDetail";
import Login from "../components/Login";
import Registro from "../components/Registro";
import Favorites from "../components/Favorites";
import { useEffect, useState } from "react";
import Loader from "../components/Loader";
import { getAuth, onAuthStateChanged } from "firebase/auth";
import { LoginRoutes } from "./LoginRoutes";

function App() {
  const [checking, setChecking] = useState(true);
  const [isLogin, setIsLogin] = useState(false);

  useEffect(() => {
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {
      if (user?.uid) {
        setIsLogin(true);
        const { uid, displayName, photoURL } = user;
        const sesion = {  uid, displayName, photoURL };
        localStorage.setItem("userPoke", JSON.stringify(sesion));
      } else {
        setIsLogin(false);
        localStorage.removeItem("userPoke");
      }
      setChecking(false);
    });
  }, [setChecking, setIsLogin]);

  if (checking) {
    return <Loader />;
  }
  return (
    <div className="App">
      <HashRouter>
        <Navbarapp isLogin={isLogin} />
        <Routes>
          <Route path="/" element={<ListCard />} />
          <Route path="/pokemon/:id" element={<CardDetail />} />        
          <Route path="/favorites" element={<Favorites />} />

          <Route
            path="/login"
            element={
              <LoginRoutes isAuthenticated={isLogin}>
                <Login />
              </LoginRoutes>
            }
          />
          <Route
            path="/registro"
            element={
              <LoginRoutes isAuthenticated={isLogin}>
                <Registro />
              </LoginRoutes>
            }
          />

          
        </Routes>
      </HashRouter>
    </div>
  );
}

export default App;
