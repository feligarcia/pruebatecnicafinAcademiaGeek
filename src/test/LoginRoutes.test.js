import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { LoginRoutes } from "../routes/LoginRoutes";
import Login from "../components/Login";
import Registro from "../components/Registro";


describe("Validar rutas con inicio de sesion", () => {
  test("Validar render de <LoginRoutes /> con <Login />", () => {
    const isLogin = true;
    const wrapper = shallow(
      <LoginRoutes isAuthenticated={isLogin} children={<Login />} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Validar render de <LoginRoutes /> con <Registro />", () => {
    const isLogin = true;
    const wrapper = shallow(
        <LoginRoutes isAuthenticated={isLogin} children={<Registro />} />
      );
    expect(wrapper).toMatchSnapshot();
  });
});
