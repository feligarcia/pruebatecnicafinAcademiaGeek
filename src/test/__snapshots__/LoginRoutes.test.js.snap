// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar rutas con inicio de sesion Validar render de <LoginRoutes /> con <Login /> 1`] = `<Login />`;

exports[`Validar rutas con inicio de sesion Validar render de <LoginRoutes /> con <Registro /> 1`] = `<Registro />`;
